annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/BaseJsonldRest.java

```diff
@@ -309,25 +301,22 @@ protected Agent buildAgent(String id) {
      * @return
      * @throws HttpException
      */
-    protected ResponseEntity<String> getAnnotationById(String identifier, String profileStr, String language)
+    protected ResponseEntity<String> getAnnotationById(long identifier, String profileStr, String language)
 	    throws HttpException {
 
 	try {
 
-	    // 3. Retrieve an annotation based on its identifier;
-	    AnnotationId annoId = new BaseAnnotationId(getConfiguration().getAnnotationBaseUrl(), identifier);
-
 	    // 4. If annotation doesn’t exist respond with HTTP 404 (if provided
 	    // annotation id doesn’t exists )
 	    // 4.or 410 (if the user is not allowed to access the annotation);
-	    Annotation annotation = getAnnotationService().getAnnotationById(annoId, null, true);
+	    Annotation annotation = getAnnotationService().getAnnotationById(identifier, null, true);
 
 	    SearchProfiles searchProfile = SearchProfiles.getByStr(profileStr);
 	    // will update body if dereference profile is used
 	    getAnnotationService().dereferenceSemanticTags(annotation, searchProfile, language);
 
-	    JsonLd annotationLd = new AnnotationLdSerializer(annotation);
-	    String jsonLd = annotationLd.toString(4);
+	    annotationLdSerializer.setAnnotation(annotation);
```

Comment:
annotationSerilizer is statefull, therefore a new object must be created for each request